socketmaster.go

```diff
@@ -30,9 +30,12 @@ func handleSignals(processGroup *ProcessGroup, c <-chan os.Signal, startTime int
 					time.Sleep(time.Duration(startTime) * time.Millisecond)
 				}
 
-				// A possible improvement woud be to only swap the
-				// process if the new child is still alive.
-				processGroup.SignalAll(syscall.SIGTERM, process)
+				if len(processGroup.set.set) > 1 {
```

Comment:
There is a concurrency issue because the access to the set bypasses the mutex. If you add a Len() method on the processSet it should be alright.